lib/LedgerSMB/Scripts/payment.pm

```diff
@@ -1216,10 +1219,18 @@ for (my $i=1 ; $i <= $request->{overpayment_qty}; $i++) {
      # Now we split the account selected options, using the namespace the if statement
      # provides for us.
      $request->{"overpayment_topay_$i"} = LedgerSMB::PGNumber->from_input($request->{"overpayment_topay_$i"});
-     $request->{"overpayment_account_$i"} =~ /^(\d+)--*/;
-     my $id = $1;
-     $request->{"overpayment_cash_account_$i"} =~ /^(\d+)--*/;
-     my $cashid = $1;
+     my $id;
+     if ( $request->{"overpayment_account_$i"} =~ /^(\d+)--*/) {
+        $id = $1;
+     } else {
+        die "Error: Invalid data";
+     }
+     my $cashid;
+     if ( $request->{"overpayment_cash_account_$i"} =~ /^(\d+)--*/) {
+        $cashid = $1;
+     } else {
+        die "Error: Invalid data";
```

Comment:
Actually in the current code, I'm expecting a cash account for overpayments to be selected. *However* I have a change planned where the "overpayments" table isn't shown when no overpayment accounts have been set up (because it can't be used anyway). When I implement that change, then this value too *can* be `undef`.,